hazelcast/src/main/java/com/hazelcast/jet/impl/JobRepository.java

```diff
@@ -170,25 +170,18 @@
     private final HazelcastInstance instance;
     private final ILogger logger;
 
-    private final IMap<Long, JobRecord> jobRecords;
-    private final IMap<Long, JobExecutionRecord> jobExecutionRecords;
-    private final IMap<Long, JobResult> jobResults;
-    private final IMap<Long, List<RawJobMetrics>> jobMetrics;
-    private final IMap<String, SnapshotValidationRecord> exportedSnapshotDetailsCache;
-    private final FlakeIdGenerator idGenerator;
+    private IMap<Long, JobRecord> jobRecords;
```

Comment:
I thought we could use `ConcurrentMemoizingSupplier`, that way you don't need the getter methods (but need `.get()` everywhere).

Reply:
good idea, it should solve the hafl-initialization issue viliam mentioned